### PR TITLE
add WeightSize() for linear_no_bias, rbf, constant

### LINKED ISSUE #2590: `WeightSetVisitor` and `WeightSizeVisitor` should calculate the size of the layer.
#### Issue description
`WeightSetVisitor` and `WeightSizeVisitor` should calculate the size of each layer by using the parameters like `inSize` and `outSize` but currently they are not doing same. It should be different for each layer.
For Linear Layer it will be `inSize * outSize + outSize` .
Note:- While adding implementation of WeigthSize please look at constructor to get the information about the size of weight matrix of that layer.
Related to https://github.com/mlpack/mlpack/pull/2324#issuecomment-679448653

 A list of all the layers -
 
  * [x]  add.hpp
  * [ ]  add_merge.hpp
  * [x]  adaptive_max_pooling.hpp
  * [x]  adaptive_mean_pooling.hpp
  * [ ]  alpha_dropout.hpp
  * [x]  atrous_convolution.hpp
  * [ ]  base_layer.hpp
  * [x]  batch_norm.hpp
  * [ ]  bilinear_interpolation.hpp
  * [x]  concat.hpp
  * [ ]  concat_performance.hpp
  * [ ]  concatenate.hpp
  * [ ]  constant.hpp
  * [x]  convolution.hpp
  * [x]  dropconnect.hpp
  * [ ]  dropout.hpp
  * [ ]  elu.hpp
  * [x]  fast_lstm.hpp
  * [ ]  flexible_relu.hpp
  * [ ]  glimpse.hpp
  * [ ]  gru.hpp
  * [ ]  hard_tanh.hpp
  * [ ]  highway.hpp
  * [ ]  join.hpp
  * [ ]  layer.hpp
  * [ ]  layer_norm.hpp
  * [ ]  layer_traits.hpp
  * [ ]  layer_types.hpp
  * [ ]  leaky_relu.hpp
  * [x]  linear.hpp
  * [ ]  linear_no_bias.hpp
  * [ ]  log_softmax.hpp
  * [ ]  lookup.hpp
  * [x]  lstm.hpp
  * [x]  max_pooling.hpp
  * [x]  mean_pooling.hpp
  * [ ]  minibatch_discrimination.hpp
  * [x]  multihead_attention.hpp
  * [x]  multiply_constant.hpp
  * [x]  multiply_merge.hpp
  * [x]  noisylinear.hpp
  * [ ]  parametric_relu.hpp
  * [ ]  positional_encoding.hpp
  * [ ]  recurrent.hpp
  * [ ]  recurrent_attention.hpp
  * [ ]  reinforce_normal.hpp
  * [ ]  reparametrization.hpp
  * [ ]  radial_basis_function.hpp
  * [ ]  select.hpp
  * [x]  sequential.hpp
  * [x]  softmax.hpp
  * [ ]  spatial_dropout.hpp
  * [ ]  subview.hpp
  * [x]  transposed_convolution.hpp
  * [ ]  vr_class_reward.hpp
  * [ ]  c_relu.hpp
  * [x]  weight_norm.hpp
  * [ ]  hardshrink.hpp
  * [ ]  celu.hpp
  * [ ]  softshrink.hpp

### PR DESCRIPTION
**Pull request by onikolskyy**
add WeightSize() for linear_no_bias, rbf, constant

Fixes #2590

### CODE REVIEW

**Comment by birm** [comment]
I hope you don't mind me editing the title of this PR :)

**Review by rcurtin — APPROVED** [review]
Looks good to me.  Thanks!  :+1:  (and please do use more descriptive PR titles, it makes it way easier to sort through the list to know what the PR is about)
